### DISCUSSION TITLE
Is it Possible to save a variable framerate stream as a constant framerate when using -o?

### DISCUSSION THREAD

**Discussion by sneakwish** [streamlink/streamlink#4541] [Q&A] [answered]
I use streamlink to record streams, but some are variable framerate, so I have to use handbrake to convert it to constant framerate.
Is it possible to record a variable framerate stream in constant framerate?

I use the latest version of streamlink and python on windows.

This is the command I use:
`streamlink https://play.afreecatv.com/somestream/ original --afreeca-username xxx --afreeca-password xxx  --afreeca-purge-credentials -o "({time:%d-%m-%Y) - [%H.%M.%S]}) {url}.ts" --stream-segment-attempts 999 --http-header "User-Agent=Mozilla/5.0" --stream-segment-threads 10 --hls-live-restart`

**Comment by bastimeyer** [comment, marked as answer]
This requires re-encoding the video stream. You can pipe the output of streamlink directly into ffmpeg and re-encode the stream on-the-fly instead of writing it to the filesystem first. This has been explained more than a dozen times on the issue tracker and discussion forum.

See #4489 for example...
https://streamlink.github.io/cli.html#cmdoption-stdout
